commands/keys.md

```diff
@@ -26,6 +26,12 @@ Supported glob-style patterns:
 
 Use `\` to escape special characters if you want to match them verbatim.
 
+Note that when enabling `Redis cluster-mode`, using a pattern with hash tag that implies a specific slot can save complexity.
+Because any key matching the pattern will be in the same slot.
+Redis would only iterate over keys in the slot, rather than the whole database.
```

Comment:
There is something strange about the formulation "can save complexity". I'm trying to make it clearer and also comparing to #2646. Is this better?

```suggestion
When using [Redis Cluster](/docs/management/scaling/), the search is optimized for patterns that imply a single slot.
If a pattern can only match keys of one slot,
Redis only iterates over keys in that slot, rather than the whole database,
when searching for keys matching the pattern.
```